> Would it also work to declare the destructor then never implement it? We should never be destroying a SameSizeAsNode.

As noted in [Bug 278282](https://bugs.webkit.org/show_bug.cgi?id=278282), this is about fixing the static assert in `CheckedRef.h` firing during builds with newer clang:

```
WebKitBuild/Release/usr/local/include/wtf/CheckedRef.h:309:48: error: no type named 'WTFDidOverrideDeleteForCheckedPtr' in 'WebCore::SameSizeAsNode'
  309 |         static_assert(std::is_same<typename T::WTFDidOverrideDeleteForCheckedPtr, int>::value, "Objects that use CanMakeCheckedPtr must use WTF_OVERRIDE_DELETE_FOR_CHECKED_PTR");
      |                                    ~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from WebKitBuild/Release/DerivedSources/WebCore/unified-sources/UnifiedSource170.cpp:8:
Source/WebCore/dom/Node.cpp:111:8: note: in instantiation of member function 'WTF::CanMakeCheckedPtr<WebCore::SameSizeAsNode>::~CanMakeCheckedPtr' requested here
  111 | struct SameSizeAsNode : EventTarget, CanMakeCheckedPtr<SameSizeAsNode> {
      |        ^
1 error generated.
```

--- a/Source/WebCore/dom/Node.cpp
+++ b/Source/WebCore/dom/Node.cpp
@@ -109,6 +109,9 @@ WTF_MAKE_COMPACT_TZONE_OR_ISO_ALLOCATED_IMPL(Node);
 using namespace HTMLNames;
 
 struct SameSizeAsNode : EventTarget, CanMakeCheckedPtr<SameSizeAsNode> {
+    WTF_MAKE_FAST_ALLOCATED;
+    WTF_OVERRIDE_DELETE_FOR_CHECKED_PTR(SameSizeAsNode);
+public:
 #if ASSERT_ENABLED
     uint32_t m_isAllocatedMemory;
     bool inRemovedLastRefFunction;

